bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/ScanAndCompareGarbageCollector.java

```diff
@@ -166,14 +168,19 @@ public void gc(GarbageCleaner garbageCleaner) {
                     if (!ledgersInMetadata.contains(bkLid)) {
                         if (verifyMetadataOnGc) {
                             int rc = BKException.Code.OK;
+                            boolean isTimeOut = false;
                             try {
-                                result(ledgerManager.readLedgerMetadata(bkLid));
+                                result(ledgerManager.readLedgerMetadata(bkLid), this.conf.getZkTimeout(),
+                                        TimeUnit.MILLISECONDS);
                             } catch (BKException e) {
                                 rc = e.getCode();
+                            } catch (Exception e) {
+                                isTimeOut = true;
+                                LOG.warn("Failed to read ledger-metadata for {} from zk {}", bkLid, e.getMessage());
                             }
-                            if (rc != BKException.Code.NoSuchLedgerExistsException) {
+                            if (isTimeOut || rc != BKException.Code.NoSuchLedgerExistsException) {
```

Comment:
the comment here doesn't seem to be right to me for `isTimeout` case. Why not just move line 182 to line 176 in `catch BKException` block?

Reply:
sure, i have fixed it. I also realized that unfortunately I have made a mistake in timeunit of zkTimeout in ##1937 where caller was passing msec but ZkUtils at receiver was expecting into seconds. So, I have also made the change to accept zk-timeout in ms.